test/js/bun/test/expect.test.js

```diff
@@ -3283,6 +3283,15 @@ describe("expect()", () => {
     expect({}).not.toBeInteger();
   });
 
+  test("toBeObject()", () => {
```

Comment:
Could we add some more test cases here? e.g. `class A {}`, `[]`, `new Set()`, etc